liquibase-standard/src/main/java/liquibase/changelog/filter/ShouldRunChangeSetFilter.java

```diff
@@ -71,10 +71,10 @@ public ChangeSetFilterResult accepts(ChangeSet changeSet) {
 
 
     protected boolean checksumChanged(ChangeSet changeSet, RanChangeSet ranChangeSet) {
-        if (ranChangeSet.getLastCheckSum() == null || changeSet.getStoredCheckSum() == null) {
+        if (ranChangeSet.getLastCheckSum() == null) {
             return false;
         }
-        return !changeSet.generateCheckSum(ChecksumVersions.enumFromChecksumVersion(changeSet.getStoredCheckSum().getVersion()))
+        return !changeSet.generateCheckSum(ChecksumVersion.enumFromChecksumVersion(ranChangeSet.getLastCheckSum().getVersion()))
```

Comment:
There was a test failure that forced me to review it. Let me see:
We have the ranchangeset from db and the changeset loaded from the xml, and we want to check if it changed.  I'm updating the comments to:
```
 protected boolean checksumChanged(ChangeSet changeSet, RanChangeSet ranChangeSet) {
        if (ranChangeSet.getLastCheckSum() == null) {
            // we don't have a last checksum to compare to, so we are not able to compare both. This may be the case of a 
            // new changeset or after a clearChecksums execution, so we return false as we are unable to test it
            return false;
        }
        // generate the checksum for the current changeset loaded from changelog files using the same checksum version
       // that we have at database so we are able to compare them
        return !changeSet.generateCheckSum(ChecksumVersion.enumFromChecksumVersion(ranChangeSet.getLastCheckSum().getVersion()))
                .equals(ranChangeSet.getLastCheckSum());
    }
```